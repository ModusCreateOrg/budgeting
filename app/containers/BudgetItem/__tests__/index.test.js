import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BudgetItem } from '../';

// mock nested component
jest.mock('components/DonutChart');

describe('BudgetItem', () => {
  const props = {
    transactions: [
      { id: 1, description: 'toiletries', categoryId: '20', value: 20 },
      { id: 2, description: 'drinks', categoryId: '10', value: -10 },
    ],
    match: { params: { id: '1' } },
    totalBudget: 30,
    history: { goBack: jest.fn() },
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders nothing', () => {
    const tree = renderer.create(<BudgetItem match={props.match} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders react element', () => {
    const tree = renderer.create(<BudgetItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should go back to previous page', () => {
    const newProps = { ...props, match: { params: { id: '2' } } };
    const component = shallow(<BudgetItem {...newProps} />);

    component.find('button[children="Go Back"]').simulate('click');

    expect(newProps.history.goBack).toHaveBeenCalled();
  });

  describe('getTransactionAndChartData', () => {
    it('should fetch new data and update state when id param changes', () => {
      const component = shallow(<BudgetItem {...props} />);
      const oldState = component.instance().state;
      const match = { params: { id: '2' } };

      component.setProps({ match });

      expect(component.instance().state).not.toEqual(oldState);
    });

    it('should get transaction and chart data on `componentWillMount`', () => {
      const spy = jest.spyOn(BudgetItem.prototype, 'getTransactionAndChartData');
      const component = shallow(<BudgetItem {...props} />);

      expect(spy).toHaveBeenCalled();
    });
  });
});
