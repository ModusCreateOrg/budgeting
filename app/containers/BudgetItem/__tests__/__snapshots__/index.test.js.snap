// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BudgetItem renders nothing 1`] = `null`;

exports[`BudgetItem renders react element 1`] = `
<section>
  <div
    className="detailsWrapper"
  >
    <div
      className="buttonWrapper"
    >
      <button
        className="backBtn"
        onClick={[Function]}
        type="button"
      >
        Go Back
      </button>
    </div>
    <div
      className="itemDetails"
    >
      <div>
        <h3
          className="m0"
        >
          <span>
            toiletries - 
          </span>
          <span
            className="pos"
          >
            $20.00
          </span>
        </h3>
        <p
          className="pos"
        >
          + 67%
        </p>
      </div>
    </div>
  </div>
</section>
`;
